cmd/restic/cmd_rebuild_index.go

```diff
@@ -97,7 +96,6 @@ func rebuildIndex(opts RebuildIndexOptions, gopts GlobalOptions, repo *repositor
 			if !ok || size != packSize {
 				// Pack was not referenced in index or size does not match
 				packSizeFromList[id] = packSize
-				removePacks.Insert(id)
```

Comment:
Shouldn't rebuild-index completely remove a broken index entry for a pack file?

Reply:
You are right - I had `!ok` case in mind.
I make a new PR for the bugfix, so I can take my time on the code simplifications...

Reply:
is corrected now.